### DISCUSSION TITLE
KaTeX Automatic Numbering

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#6039] [Q&A] [answered]
KaTeX nicely uses automatic numbering for some environments! One issue is when there are several tiddlers having KaTeX equations and you open those tiddlers in the story river the equation numbers are updated and based on the order of tiddlers the number changes!

The question is how one can force reset numbering on each tiddler!

I know I can reset numbering by wrapping `$$...$$ into a div for example:

```
<style>
.main  { counter-reset: katexEqnNo; }
</style>
<div class=main>

tiddler contents....
.
.
</div>
```

Does TW tiddler have any specific class to be used here? I mean a unique class tell us a tiddler has been started and there was no need to use an extra class like the above

**Comment by kookma** [comment, marked as answer]
I did some experiments and it seems, a stylesheet (a tiddler tagged with `$:/tags/Stylesheet`) as below works

```
.tc-tiddler-frame { counter-reset: katexEqnNo; }
```

do you recommend this solution?

**Reply by pmario** [reply]
Yes. `.tc-tiddler-body` is the DIV that separates the tiddler-body from the  other tiddler elements. eg: title, subtitle, tags and so on. .. 

tiddler-frame `.tc-tiddler-frame`. contains all the elements that make a tiddler in the story-river.

**Reply by Jermolene** [reply]
Sounds reasonable - @pmario might you be able to create a PR?

**Reply by pmario** [reply]
>Sounds reasonable - @pmario might you be able to create a PR?

yes.

@kookma .. Are there any other elements, that need resetting that way?

**Reply by pmario** [reply]
@kookma You may add more info at the issue: https://github.com/Jermolene/TiddlyWiki5/issues/6041 So I can create a PR

**Reply by kookma** [reply]
@pmario - I asked the question from KaTeX author (see https://github.com/KaTeX/KaTeX/discussions/3265). It seems the `katexEqnNo` is the counter!

One can reset number by sections e.g. headings (like in an article), so he/she can customize it using

```
h1 { counter-reset: katexEqnNo; }
```

So, I think the `katexEqnNo` is the only counter to be reset! It controls numbering for equation, align, aligned, gather,...
I have created some test case and docs here ([KaTex Automatic Numbering](https://kookma-katex.tiddlyhost.com/#Reset%20Automatic%20Numbering%20in%20Each%20Tiddler:%5B%5BReset%20Automatic%20Numbering%20in%20Each%20Tiddler%5D%5D%20%5B%5BReset%20Automatic%20Numbering%20by%20Section%5D%5D%20%5B%5BDisable%20KaTeX%20Automatic%20Numbering%5D%5D%20%5B%5BExample%20i-%20Automatic%20Numbering%5D%5D%20%5B%5BExample%20ii-%20Automatic%20Numbering%5D%5D%20%5B%5BExample%20-%20Transclude%20Tiddlers%20for%20Numbering%5D%5D))


@Jermolene - I have created a documentation wiki for KaTeX in Tiddlywiki (see https://kookma-katex.tiddlyhost.com/) I may propose to add a link in the official KaTeX Demo wiki!

**Comment by pmario** [comment]
@kookma @Jermolene  ... There's a PR: #6046
